lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -187,13 +200,8 @@ public function hydrateSet($key, $element)
     {
         $this->collection->set($key, $element);
 
-        // If _backRefFieldName is set, then the association is bidirectional
-        // and we need to set the back reference.
         if ($this->backRefFieldName && $this->association['type'] === ClassMetadata::ONE_TO_MANY) {
-            // Set back reference to owner
-            $this->typeClass->reflFields[$this->backRefFieldName]->setValue(
-                $element, $this->owner
-            );
+            $this->completeBidirectionalAssociationLink($element);
```

Comment:
The looks weird to me. Conceptually, if you're `hydrateSet`ing, you should have the back reference defined.
However, if you're doing that for a new collection owner entity, the you should not have the original entity property value defined.